Enhancement/drawer mobile

camilovegag opened the pull request:
### Closes: #42 

### Description: 
Improving our existing modal with
- [x] Mini drawer variant toggleable by clicking the hamburger menu icon on screen width 500px or bigger.
- [x] Highlight with HOPR blue color the current drawer item selected (by matching path).
- [x] Drawer will auto become mini variant on screens with 900px or less.
- [ ] (To be defined) trigger mini variant when resizing window and it goes below 900px.

mjadach-iv reviewed (changes requested):
- The humberger icon should have a round hightlight
- the drawer as it will be only used in layout component, it should go back it its original folder
- lets use the same duration for the animation of streaching content page and hiding drawer, as now they seem to be unlinked in terms of speed
-

camilovegag commented on src/future-hopr-lib-components/Layout/index.tsx:
```diff
@@ -58,6 +71,13 @@ const Layout: React.FC<{
+
+  // Set the initial state of the drawer based on the device type
+  // If it's a mobile device, set the drawer to be closed by default
+  const [openNavigationDrawer, set_openNavigationDrawer] = useState(!isMobile);
```
Changed to `openedNavigationDrawer`

camilovegag commented:
> * The humberger icon should have a round hightlight
> * the drawer as it will be only used in layout component, it should go back it its original folder
> * lets use the same duration for the animation of streaching content page and hiding drawer, as now they seem to be unlinked in terms of speed

- Updated the icon
- Moved drawer back to layout
- This was due to drawerRight changing the margin too, add the transition only to `margin-left`

mjadach-iv commented on src/future-hopr-lib-components/Layout/drawer.tsx:
```diff
@@ -1,156 +1,147 @@
+                  selected={location.pathname === `/${group.path}/${item.path}`}
+                  disabled={item.loginNeeded && !drawerLoginState?.[item.loginNeeded]}
+                  onClick={handleButtonClick}
+                  sx={{
+                    height: 48,
+                    '&.Mui-selected': {
+                      color: '#0000B4',
+                      '& .MuiTypography-root': { fontWeight: 'bold' },
+                      '& .MuiListItemIcon-root': { color: '#0000B4' },
+                    },
+                  }}
```
you are using 2 ways of styling, lets only use styled components

mjadach-iv commented on src/future-hopr-lib-components/Layout/drawer.tsx:
```diff
@@ -1,156 +1,147 @@
+                  component={Link}
+                  to={`${group.path}/${item.path}${searchParams ?? ''}`}
+                  selected={location.pathname === `/${group.path}/${item.path}`}
+                  disabled={item.loginNeeded && !drawerLoginState?.[item.loginNeeded]}
```
if a page should be disabled, it is avalible and if we click on it, the app crashes

mjadach-iv commented on src/future-hopr-lib-components/Layout/drawer.tsx:
```diff
@@ -1,156 +1,147 @@
   drawerLoginState?: {
-    node?: boolean;
-    web3?: boolean;
+    node: boolean;
+    web3: boolean;
```
With this change, we will not be able to build the app only as a Node Admin or Staking UI. Please put it back

mjadach-iv commented on .eslintrc.json:
```diff
@@ -26,15 +26,18 @@
-      "functions": "always-multiline"
-    }],
+    "comma-dangle": [
+      2,
+      {
+        "arrays": "always-multiline",
+        "objects": "always-multiline",
+        "imports": "never",
+        "exports": "always-multiline",
+        "functions": "always-multiline"
+      }
+    ],
     "import-newlines/enforce": "warn",
-    "newline-destructuring/newline": "warn",
+    "newline-destructuring/newline": ["warn", { "items": 1 }],
```
If you do changes like that in a PR that should only touch other files, then you can easily create more work and conflicts, which is the case now. For the future, please separate PRs like that so we will not have conflicts.

camilovegag replied:
Okay

mjadach-iv reviewed (changes requested):
Please resolve conflicts

camilovegag commented:
> Please resolve conflicts

Fixed
